rust/lance/src/dataset/scanner.rs

```diff
@@ -810,17 +948,27 @@ impl Scanner {
             plan = self.limit_node(plan);
         }
 
-        // Stage 5: take remaining columns / projection
-        let output_schema = self.output_schema()?;
-        let remaining_schema = output_schema.exclude(plan.schema().as_ref())?;
+        // Stage 5: take remaining columns required for projection
+        let physical_schema = self.physical_schema()?;
+        let remaining_schema = physical_schema.exclude(plan.schema().as_ref())?;
         if !remaining_schema.fields.is_empty() {
             plan = self.take(plan, &remaining_schema, self.batch_readahead)?;
         }
-        let output_arrow_schema = output_schema.as_ref().into();
+
+        // Stage 6: physical projection -- reorder physical columns needed before final projection
+        let output_arrow_schema = physical_schema.as_ref().into();
```

Comment:
Isn't this redundant now? Would it be difficult to eliminate this?

Reply:
this is part of the schema mess. I wanted to keep things a bit more "simple", so I made the final projection uses the schema of the scanner before my PR.

I need this projection stage to order things into expect ordering before projecting. Once clean up how schema is exposed this can be removed.